pkg/operation/seed/seed.go

```diff
@@ -590,19 +595,25 @@ func BootstrapCluster(ctx context.Context, k8sGardenClient, k8sSeedClient kubern
 			Image:           igwImage.String(),
 			IstiodNamespace: common.IstioNamespace,
 			Annotations:     seed.LoadBalancerServiceAnnotations,
+			Ports:           []corev1.ServicePort{},
 		}
 
-		if gardenletfeatures.FeatureGate.Enabled(features.APIServerSNI) {
-			ports := []corev1.ServicePort{
-				{Name: "proxy", Port: 8443, TargetPort: intstr.FromInt(8443)},
-				{Name: "tcp", Port: 443, TargetPort: intstr.FromInt(9443)},
-			}
-
-			if gardenletfeatures.FeatureGate.Enabled(features.KonnectivityTunnel) {
-				ports = append(ports, corev1.ServicePort{Name: "tls-tunnel", Port: 8132, TargetPort: intstr.FromInt(8132)})
-			}
+		// even if SNI is being disabled, the existting ports must stay the same
+		// until all APIServer SNI resources are removed.
+		if gardenletfeatures.FeatureGate.Enabled(features.APIServerSNI) || anySNI {
+			igwConfig.Ports = append(
+				igwConfig.Ports,
+				corev1.ServicePort{Name: "proxy", Port: 8443, TargetPort: intstr.FromInt(8443)},
+				corev1.ServicePort{Name: "tcp", Port: 443, TargetPort: intstr.FromInt(9443)},
+			)
+		}
 
-			igwConfig.Ports = ports
+		if gardenletfeatures.FeatureGate.Enabled(features.APIServerSNI) &&
+			gardenletfeatures.FeatureGate.Enabled(features.KonnectivityTunnel) {
```

Comment:
Just to be sure: Do we need the same here, i.e. 
```suggestion
		if (gardenletfeatures.FeatureGate.Enabled(features.APIServerSNI) || anySNI) &&
			gardenletfeatures.FeatureGate.Enabled(features.KonnectivityTunnel) {
```